### PR TITLE
Fixes from today's telecon

### PR DESCRIPTION
**Pull request by sbird**
Here are the fixes from today's telecon for the small black hole seed mass and for the ScaleDepVelocity. There is one more fix to make address sanitizer happy.

### CODE REVIEW

File: genic/main.c [hunk 1]
```diff
@@ -43,7 +43,7 @@ int main(int argc, char **argv)
 
   petapm_init(All.BoxSize, All.Nmesh, All.NumThreads);
   /*Initialise particle spacings*/
-  const double meanspacing = All.BoxSize / All2.Ngrid;
+  const double meanspacing = All.BoxSize / DMAX(All2.Ngrid, All2.NgridGas);
```
Comment (rainwoodman) [inline genic/main.c]: This looks strange. perhaps need meanspacing for each species instead.

Comment (sbird) [reply genic/main.c]: It is only used for computing the shifts - the important thing is actually that the shift is less than one grid cell for all particles, so this is the right thing to do.

File: libgadget/fof.c [hunk 2]
```diff
@@ -796,7 +796,7 @@ fof_compile_catalogue(struct Group * group, double BoxSize, int BlackHoleInfo, M
     fof_finish_group_properties(Group, BoxSize);
 
     int64_t TotNids;
-    MPI_Allreduce(&Ngroups, &TotNgroups, 1, MPI_UINT64, MPI_SUM, Comm);
+    sumup_large_ints(1, &Ngroups, &TotNgroups);
```
Comment (rainwoodman) [inline libgadget/fof.c]: It's cleaner to use TotNgroups = Ngroups and MPI_IN_PLACE.
Most use cases of sumup_large_ints can be replaced by this.

Comment (sbird) [reply libgadget/fof.c]: Ah yes - that way the right communicator is used. Thanks.

**Pull request merged**